apps/agent/test/agent_shared.js

```diff
@@ -734,11 +734,14 @@ module.exports = (agentName, { accounts, artifacts, web3 }) => {
         context('> Designated signer', () => {
           const ethSign = async (hash, signer) => {
             const packedSig = await web3.eth.sign(hash, signer)
+            const chainId = await web3.eth.net.getId()
+            // If we detect buidlerevm we don't use ganache encoding
+            const offset = chainId === 31337 ? 0 : 27
```

Comment:
Ohhh interesting, are there docs for this? I'd like to leave a few more comments :).

We may also want to push something like this into the contract-helpers (we can do it later though)

Reply:
This was pointing out by Pato from buidler. He told me that ganache encodes the v value of the signature as 0 or 1, while it should be 27 or 28. But buidler uses the same values as geth.

Related discussions about the topic:
- https://github.com/ethereumjs/ethereumjs-util/pull/67#issuecomment-274926471
- https://github.com/trufflesuite/ganache-core/issues/555

Reply:
Interesting, let's leave these in the inline code!